edb/edgeql/compiler/stmt.py

```diff
@@ -75,6 +75,16 @@ def compile_SelectQuery(
             sctx.partial_path_prefix = ctx.partial_path_prefix
             stmt.implicit_wrapper = True
 
+            # We need to preserve view_rptr if this SELECT is just
+            # an implicit wrapping of a single DISTINCT, because otherwise
+            # using a DISTINCT to satisfy link multiplicity requirement
+            # will kill the link properties.
+            #
+            # This includes problems with initializing the schema itself.
+            if (isinstance(expr.result, qlast.UnaryOp) and
+                    expr.result.op == 'DISTINCT'):
```

Comment:
This is a general problem of nesting computable link properties. Hardcoding for `DISTINCT` specifically is not going to help much.  I have some considerations on how to refactor link property derivation in general, the idea is to attach them to the derived target type instead of trying to pass through the remote pointer trough context.